src/main/webapp/app/shared/util/core-evidence-submission/core-evidence-submission-utils.ts

```diff
@@ -101,3 +101,148 @@ export function getNewPriorities(list: Treatment[], unapprovedUuids: string | st
   });
   return priorities;
 }
+
+type CollectUUIDsArgs = (
+  | {
+      type: 'mutation';
+      obj: Mutation;
+    }
+  | {
+      type: 'tumor';
+      obj: Tumor;
+    }
+  | {
+      type: 'TI';
+      obj: TI;
+    }
+  | {
+      type: 'treatment';
+      obj: Treatment;
+    }
+) & {
+  uuids?: string[];
+  uuidType?: 'insideOnly' | 'evidenceOnly' | 'sectionOnly';
+};
+export function collectUUIDs({ type, obj, uuids = [], uuidType }: CollectUUIDsArgs): string[] {
```

Comment:
Do you like how this method is written? We can also break them up to collectMutationUuids etc.

Reply:
I like it. This kind of style is becoming more and more popular. I will admit it looks better when you have [pattern matching support](https://dev.java/learn/pattern-matching/) built into the language.

Reply:
I was ahead of JAVA 16 :) Could you also move the comment from the legacy to here as well? I think they are pretty useful. And update it accordingly.